### PR TITLE
Monorepo: Enable linters in advisory mode

### PR DESCRIPTION
**Pull request by mandarjog**
**What this PR does / why we need it**:
 Enable linters in advisory mode
```release-note
NONE
```

### CODE REVIEW

File: bin/linters.sh [hunk 1]
```diff
@@ -44,15 +52,15 @@ docker run\
   --enable=varcheck\
   --enable=vet\
   --enable=vetshadow\
-  ./...
+  ./... || true
 echo 'linters OK'
```
Comment (rshriram) [inline bin/linters.sh]: You mean &&?

Comment (rshriram) [reply bin/linters.sh]: Ah ignore

File: .circleci/config.yml [hunk 2]
```diff
@@ -36,6 +36,7 @@ jobs:
           keys:
             - bazel-cache-{{ checksum "WORKSPACE" }}
       - run: make test
+      - run: bin/linters.sh
```
Comment (rshriram) [inline .circleci/config.yml]: You need to send skip init here as well. We are using pre cached code.

**Comment by yutongz** [comment]
Well, do we think advisory mode is a better idea. I understand the pain when the tests fail because of linter, but the fact is we already saw several linter issues brought into master and it will become a big issue eventually.

**Comment by mandarjog** [comment]
After all linter errors are fixed, and we are very close to it now, please remove the advisory mode.
From then on linterns should start failing.
For example security had tons of linter warnings.

File: bin/linters.sh [hunk 3]
```diff
@@ -44,15 +52,15 @@ docker run\
...
 
 echo 'Checking licences'
-bin/check_license.sh
+bin/check_license.sh || true
```
Comment (ldemailly) [inline bin/linters.sh]: what does it do to run these but ignore the results?

**Comment by ldemailly** [comment]
adding excludes (or better, fixing the errors) seems better than completely turning off the gating

I guess same philosophy difference wrt e2e tests

**Pull request merged**